cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -612,7 +688,84 @@ public SchemasInfoOfClusterResponse getAllSchemasInfoFromCluster(
+        clusterIdentification,
+        schemasVersionsStorageKey,
+        schemasInfoOfClusterResponse);
+
     return schemasInfoOfClusterResponse;
   }
+
+  private static void updateCache(
+      String bootstrapServers,
+      KafkaSupportedProtocol protocol,
+      String clusterIdentification,
+      String schemasVersionsStorageKey,
+      SchemasInfoOfClusterResponse schemasInfoOfClusterResponse) {
+    schemasInfoOfClusterResponseMap.put(schemasVersionsStorageKey, schemasInfoOfClusterResponse);
+    SchemaInfoCacheKeySet schemaInfoCacheKeySet = new SchemaInfoCacheKeySet();
+    schemaInfoCacheKeySet.setProtocol(protocol);
+    schemaInfoCacheKeySet.setClusterIdentification(clusterIdentification);
+    schemaInfoCacheKeySet.setBootstrapServers(bootstrapServers);
+    schemasInfoCacheKeySetMap.put(schemasVersionsStorageKey, schemaInfoCacheKeySet);
+  }
+
+  private SchemasInfoOfClusterResponse handleInterimUpdatesOnSchemas(
+      String bootstrapServers,
+      KafkaSupportedProtocol protocol,
+      String clusterIdentification,
+      SchemaCacheUpdateType updateType,
+      String topicName,
+      String schemasVersionsStorageKey) {
+    SchemasInfoOfClusterResponse schemasInfoOfClusterResponse =
+        schemasInfoOfClusterResponseMap.get(schemasVersionsStorageKey);
+    List<SchemaInfoOfTopic> schemaInfoList =
+        schemasInfoOfClusterResponse.getSchemaInfoOfTopicList();
+    Optional<SchemaInfoOfTopic> optionalSchemaInfoOfTopic =
+        schemaInfoList.stream()
+            .filter(schemaInfoOfTopic -> schemaInfoOfTopic.getTopic().equals(topicName))
+            .findFirst();
+
+    if (updateType == SchemaCacheUpdateType.CREATE) {
+      Set<Integer> schemaVersions =
+          getSchemaVersions(bootstrapServers, topicName, protocol, clusterIdentification);
+
+      if (optionalSchemaInfoOfTopic.isPresent()) {
+        optionalSchemaInfoOfTopic.get().setSchemaVersions(schemaVersions);
+      } else {
+        SchemaInfoOfTopic schemaInfoOfTopic = new SchemaInfoOfTopic();
+        schemaInfoOfTopic.setTopic(topicName);
+        schemaInfoOfTopic.setSchemaVersions(schemaVersions);
+        schemaInfoList.add(schemaInfoOfTopic);
+      }
+    } else if (updateType == SchemaCacheUpdateType.DELETE) {
+      optionalSchemaInfoOfTopic.ifPresent(schemaInfoList::remove);
+    }
+
+    return schemasInfoOfClusterResponse;
+  }
+
+  @Async("resetSchemaCacheTaskExecutor")
+  @Scheduled(
+      cron = "${klaw.schemainfo.cron.expression:0 0 0 * * ?}",
+      zone = "${klaw.schemainfo.cron.expression.timezone:UTC}")
+  public void resetSchemaCacheScheduler() {
+    for (String schemasVersionsStorageKey : schemasInfoCacheKeySetMap.keySet()) {
+      SchemaInfoCacheKeySet schemaInfoCacheKeySet =
+          schemasInfoCacheKeySetMap.get(schemasVersionsStorageKey);
+      schemasInfoOfClusterResponseMap.remove(schemasVersionsStorageKey);
+      loadAllSchemasInfoFromCluster(
+          schemaInfoCacheKeySet.getBootstrapServers(),
```

Comment:
I think you can also run the loadAllSchemasInfoFromCluster from here althogh I see this isnt async so it may cause an issue to do so.

Reply:
this scheduler process is already running in async thread, we can skip